src/test/scala/markatta/futiles/CancellableSpec.scala

```diff
@@ -0,0 +1,59 @@
+package markatta.futiles
+
+import java.util.concurrent.atomic.AtomicBoolean
+import scala.concurrent.{CancellationException, ExecutionException}
+
+class CancellableSpec extends Spec {
+  describe("The cancellable utility") {
+
+    describe("without cancellation") {
+
+      it("works as a normal Future") {
+        val cancellable = Cancellable {
+          ()
+        }
+
+        cancellable.future.futureValue shouldEqual ()
+      }
+
+      it("throws an Exception correctly") {
+        val cancellable = Cancellable {
+          throw new IllegalArgumentException
+        }
+
+        val exception = cancellable.future.failed.futureValue
+        exception shouldBe an[ExecutionException]
+        exception.getCause shouldBe an[IllegalArgumentException]
```

Comment:
So this is different behavior from a standard `Future`, with a normal `Future` if you throw an `Exception` you get back the same exception, i.e.

```scala
val f = Future {
    throw new IllegalArgumentException
  }
f.failed.futureValue shouldBe an [IllegalArgumentException]
```

I can either document this or modify the implementation so that we propagate the inner exception rather the original `ExecutionException`

Reply:
Hmm, I think unwrapping would be nicest, relatively unlikely that user code throws execution exception?

Reply:
Done